### PR TITLE
[PATCH v4] linux-gen: packet: inline fast path function implementations

### PR DESCRIPTION
**Pull request by MatiasElo**

### CODE REVIEW

**Review by TuomasTaipale — APPROVED** [review]

File: platform/linux-generic/include/odp/api/plat/packet_inlines.h [hunk 1]
```diff
@@ -423,6 +424,18 @@ _ODP_INLINE uint32_t odp_packet_buf_size(odp_packet_buf_t pkt_buf)
...
 }
 
+_ODP_INLINE void *odp_packet_buf_head(odp_packet_buf_t pkt_buf)
+{
+	odp_pool_t pool = _odp_pkt_get(pkt_buf, odp_pool_t, pool);
+	const uint32_t head_offset = _odp_pool_get(pool, uint32_t, ext_head_offset);
+
+	/* Check that pool is external */
+	if (odp_unlikely(!head_offset))
+		return ODP_PACKET_BUF_INVALID;
```
Comment (psavol) [inline platform/linux-generic/include/odp/api/plat/packet_inlines.h:438]: Should return NULL on failure, as return type is pointer (not odp_packet_buf_t).

File: platform/linux-generic/include/odp/api/plat/packet_inlines.h [hunk 2]
```diff
@@ -443,6 +444,17 @@ _ODP_INLINE uint32_t odp_packet_buf_data_offset(odp_packet_buf_t pkt_buf)
 			(uintptr_t)odp_packet_buf_head(pkt_buf);
 }
 
+_ODP_INLINE void odp_packet_buf_data_set(odp_packet_buf_t pkt_buf, uint32_t data_offset,
+					 uint32_t data_len)
+{
+	uint8_t *head = (uint8_t *)odp_packet_buf_head(pkt_buf);
```
Comment (psavol) [inline platform/linux-generic/include/odp/api/plat/packet_inlines.h:452]: Is this cast from void* to uint8_t* needed?

File: platform/linux-generic/include/odp/api/plat/packet_inlines.h (hunk 1 above)
Comment (MatiasElo) [reply platform/linux-generic/include/odp/api/plat/packet_inlines.h:438]: Fixed in V2.

File: platform/linux-generic/include/odp/api/plat/packet_inlines.h (hunk 2 above)
Comment (MatiasElo) [reply platform/linux-generic/include/odp/api/plat/packet_inlines.h:452]: It's required for C++ compatibility.

**Review by psavol — APPROVED** [review]

**Pull request merged**